Monika After Story/game/event-rules.rpy

```diff
@@ -714,6 +715,63 @@ init -1 python:
             return ev.rules.get(EV_RULE_PRIORITY, MASPriorityRule.DEF_PRIORITY)
 
 
+    class MASProbabilityRule(object):
+        """
+        Static class used to create probability rules.
+
+        Priority rules are just integers that determine the probability of something being selected.
+
+        Probabilities lie between 1 and 10, with a default of 1
+        """
+        DEF_PROBABILITY = 1
+
+        @staticmethod
+        def create_rule(probability, ev=None):
+            """
+            IN:
+                probability - the probability to set.
+                    If None is passed in, we use the default priority value.
+                    NOTE: If it is outside the bounds, it will be formatted to fit the bounds
+                    NOTE: 100 probability does NOT mean a 100% chance of selecting something
+                ev - Event to add this rule to. This will replace existing
+                    rules of the same key.
+                    (Default: None)
+            """
+            if probability is None:
+                probability = MASProbabilityRule.DEF_PROBABILITY
+
+            elif probability > 10:
+                probability = 10
+
+            elif probability < 1:
+                probability = 1
+
+            if type(probability) is not int:

```

Comment:
```suggestion
            if not store.mas_ev_data_ver._verify_int(probability, allow_none=False):
```
I would actually put this in for the None check and just default the default probability if given an invalid type. Maybe log to the `mas_log` if desired.